src/AuthenticationResponse.js

```diff
@@ -82,6 +102,37 @@ class AuthenticationResponse {
     return response
   }
 
+  /**
+   * errorResponse
+   *
+   * @param {AuthenticationResponse} response
+   *
+   * @throws {Error} If response params include the OAuth2 'error' param,
+   *   throws an error based on it.
+   *
+   * @returns {AuthenticationResponse} Chainable
+   *
+   * @todo Figure out HTTP status code (typically 400, 401 or 403)
+   *   based on the OAuth2/OIDC `error` code, probably using an external library
+   */
+  static errorResponse (response) {
+    const errorCode = response.params.error
```

Comment:
Passing things around this way creates global state.

Reply:
I don't think so, in this particular case. `response` isn't the Express `ServerResponse` object. it's a `AuthenticationResponse` instance, and the params are parsed into it.